src/main/java/org/wso2/carbon/mongodb/query/MongoDBQueryException.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.mongodb.query;
+
+/**
+ * MongoDB prepared statement exception class
+ */
+@SuppressWarnings({"unused", "WeakerAccess"})
+public class MongoDBQueryException extends Exception {
```

Comment:
Can we extend this from UserStoreException? Then we can throw this exception instead of UserStoreException.

Reply:
All the inbuilt user stores are throwing UserStoreException. Further, this exception is thrown only at places where MongoDB query related problems occur. Hence, extended the Java Exception class.